# this patch fixes the insecure tempfile usage reported in bug #235822
# it was extracted from Debian's r-base_2.7.2-1.diff patch

diff -Naur R-2.7.2/src/scripts/javareconf R-2.7.2.new/src/scripts/javareconf
--- R-2.7.2/src/scripts/javareconf	2008-03-25 08:26:44.000000000 -0400
+++ R-2.7.2.new/src/scripts/javareconf	2008-08-27 16:49:04.000000000 -0400
@@ -125,16 +125,19 @@
 javac_works='not present'
 if test -n "$JAVAC"; then
     javac_works='not functional'
-    rm -rf /tmp/A.java /tmp/A.class
-    echo "public class A { }" > /tmp/A.java
-    if test -e /tmp/A.java; then
-	if "${JAVAC}" /tmp/A.java >/dev/null; then
-	    if test -e /tmp/A.class; then
+    # edd 25 Aug 2008  use mktemp -t -d
+    #rm -rf /tmp/A.java /tmp/A.class
+    jctmpdir=`mktemp -t -d`
+    echo "public class A { }" > ${jctmpdir}/A.java
+    if test -e ${jctmpdir}/A.java; then
+	if "${JAVAC}" ${jctmpdir}/A.java >/dev/null; then
+	    if test -e ${jctmpdir}/A.class; then
 		javac_works=yes
 	    fi
 	fi
     fi
-    rm -rf /tmp/A.java /tmp/A.class
+    # rm -rf /tmp/A.java /tmp/A.class
+    rm -rf ${jctmpdir}
 fi
 if test "${javac_works}" = yes; then
     echo "Java compiler    : ${JAVAC}"
